### PR TITLE
One-Click Postage: Don't fallback to main blog on failure

### PR DESCRIPTION
**Pull request by marcustyphoon**
This removes a mystifying behavior in which One-Click Postage would, on any failure to post, do its single retry attempt with the target blog changed to the user's main.

This is a common complaint when the post author has the user's sideblog blocked. Silently reblogging to the wrong location in this case makes zero sense.

I'm sure there was a reason for this originally (maybe accessing sideblogs was unreliable?) but I can imagine no case in which the post winding up in the wrong place was a good fallback. If you can't do what the user wants, just fail the action. Simple as that.

(I didn't check if this merge conflicts with #2137, but if it does I'll fix the conflict after either is merged. If it matters, of course—I'm not willing to go out of my way to solicit further discussion.)

### CODE REVIEW

**Comment by marcustyphoon** [comment]
Tested by making a post on one of my testing accounts, blocking one of my sideblogs, and attempting to reblog the post onto that sideblog:

Screenshot 2023-10-18 at 7 39 55 PM

**Review by AprilSylph — APPROVED** [review]
Code LGTM, and I trust your testing, especially for such a straightforward change.

Up to you whether to wait for @nightpool's review before merging.

**Review by nightpool — APPROVED** [review]

**Pull request merged**
